### PR TITLE
[CBRD-23444] const db_get_string

### PR DESCRIPTION
**Pull request by valerinistor**
http://jira.cubrid.org/browse/CBRD-23444

* change `db_get_string` function signature to return `const char *`
* change type of `db_char::medium::buf` to `const char *`

Part 5 of #1656

### CODE REVIEW

File: src/query/string_opfunc.c [hunk 1]
```diff
@@ -19469,9 +19457,9 @@ db_string_reverse (const DB_VALUE * src_str, DB_VALUE * result_str)
       if (error_status == NO_ERROR)
 	{
 	  memset (res, 0, db_get_string_size (src_str) + 1);
-	  intl_reverse_string ((unsigned char *) db_get_string (src_str), (unsigned char *) res,
-			       db_get_string_length (src_str), db_get_string_size (src_str),
-			       db_get_string_codeset (src_str));
+	  intl_reverse_string (REINTERPRET_CAST (const unsigned char *, db_get_string (src_str)),
```
Comment (razvanradulescu) [inline src/query/string_opfunc.c]: you may use DB_GET_UCHAR

File: src/query/string_opfunc.c [hunk 2]
```diff
@@ -27278,7 +27269,7 @@ db_string_to_base64 (DB_VALUE const *src, DB_VALUE * result)
       return error_status;
     }
 
-  src_buf = (const unsigned char *) db_get_string (src);
+  src_buf = REINTERPRET_CAST (const unsigned char *, db_get_string (src));
```
Comment (razvanradulescu) [inline src/query/string_opfunc.c]: you may use DB_GET_UCHAR

File: src/broker/cas_execute.c [hunk 3]
```diff
@@ -5627,7 +5628,7 @@ fetch_attribute (T_SRV_HANDLE * srv_handle, int cursor_pos, int fetch_count, cha
 	  return ERROR_INFO_SET (err_code, DBMS_ERROR_INDICATOR);
 	}
 
-      class_name = db_get_string (&val_class);
+      class_name = CONST_CAST (char *, db_get_string (&val_class));
```
Comment (ribram) [inline src/broker/cas_execute.c]: todo: fix in another patch; class_name should be const char *

File: src/base/intl_support.c [hunk 4]
```diff
@@ -3113,9 +3123,9 @@ intl_identifier_upper (const char *src, char *dst)
 int
 intl_identifier_fix (char *name, int ident_max_size, bool error_on_case_overflow)
 {
-  int i, length_bytes;
+  int name_length = 0, length_bytes;
```
Comment (razvanradulescu) [inline src/base/intl_support.c]: We usually use length when referring to characters and size when referring to raw bytes.
I propose to rename :
length_bytes -> original_size
name_length  -> truncated_size

File: src/query/string_opfunc.c [hunk 5]
```diff
@@ -4732,7 +4720,7 @@ db_string_limit_size_string (DB_VALUE * src_string, DB_VALUE * result, const int
 
   if (adj_char_size > 0)
     {
-      memcpy ((char *) r, (char *) db_get_string (src_string), adj_char_size);
+      memcpy ((char *) r, db_get_string (src_string), adj_char_size);
```
Comment (ribram) [inline src/query/string_opfunc.c]:
```suggestion
      memcpy (r, db_get_string (src_string), adj_char_size);
```

File: src/query/string_opfunc.c [hunk 6]
```diff
@@ -9247,15 +9234,18 @@ qstr_position (const char *sub_string, const int sub_size, const int sub_length,
        *  try again.  This is repeated until a match is found, or
        *  there are no more comparisons to be made.
        */
-      ptr = (unsigned char *) src_string;
+      const unsigned char *sub_string_ = REINTERPRET_CAST (const unsigned char *, sub_string);
```
Comment (ribram) [inline src/query/string_opfunc.c]: use `u` or `ustr` prefix or suffix to name reinterpreted variables.

File: src/query/string_opfunc.c [hunk 7]
```diff
@@ -27369,7 +27360,7 @@ db_string_from_base64 (DB_VALUE const *src, DB_VALUE * result)
       return NO_ERROR;
     }
 
-  src_buf = (const unsigned char *) db_get_string (src);
+  src_buf = REINTERPRET_CAST (const unsigned char *, db_get_string (src));
```
Comment (ribram) [inline src/query/string_opfunc.c]: you may use DB_GET_UCHAR

File: src/query/string_opfunc.c [hunk 8]
```diff
@@ -27811,7 +27802,7 @@ db_tz_offset (const DB_VALUE * src_str, DB_VALUE * result_str, DB_DATETIME * dat
 int
 db_from_tz (DB_VALUE * time_val, DB_VALUE * tz, DB_VALUE * time_val_with_tz)
 {
-  char *timezone;
+  const char *timezone_ = NULL;
```
Comment (ribram) [inline src/query/string_opfunc.c]:
```suggestion
  const char *timezone_str = NULL;
```

File: src/broker/cas_execute.c (hunk 3 above)
Comment (valerinistor) [reply src/broker/cas_execute.c]: Ok, will fix in a separate patch

**Pull request merged**